hasktorch/src/Torch/NN.hs

```diff
@@ -107,9 +113,17 @@ data LinearSpec = LinearSpec { in_features :: Int, out_features :: Int }
 
 data Linear = Linear { weight :: Parameter, bias :: Parameter } deriving (Show, Generic)
 
+linear' :: Linear -> Tensor -> Tensor
+linear' layer input = linear input w b
+    where
+        linear input weight bias = unsafePerformIO $ (cast3 ATen.linear_ttt) input weight bias
+        w = toDependent (weight layer)
+        b = toDependent (bias layer)
+
 instance Randomizable LinearSpec Linear where
   sample LinearSpec{..} = do
-      w <- makeIndependent =<< randn' [in_features, out_features]
+      w <- makeIndependent =<< kaimingUniform' [out_features, in_features]
+      -- w <- makeIndependent =<< randn' [out_features, in_features]
```

Comment:
We should probably decide if we want to match upstream with initialization, or if we want to use some more modern choices, and apply those rules more consistently.

Reply:
Could you point to some more modern choices you have in mind?

I don't feel too strongly atm, at least pytorch's defaults are well known (untyped hasktorch optimizations seem to be well behaved for me now, though haven't ventured into hairier architectures much). If there's something better I wouldn't mind trying as long as it's not much more complex to implement.

Reply:
Well pure randn is simply a bad idea, PyTorch uses some variants of LeCun initialization, kaimingUniform is yet another choice (I think a more modern one?).